quickstart/101-vm-cluster-linux/main.tf

```diff
@@ -0,0 +1,123 @@
+  prefix = "vnet"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = random_pet.azurerm_virtual_network_name.id
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+}
+
+resource "random_pet" "azurerm_subnet_name" {
+  prefix = "sub"
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = random_pet.azurerm_subnet_name.id
+  resource_group_name  = azurerm_resource_group.rg.name
+  virtual_network_name = azurerm_virtual_network.test.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+resource "azurerm_public_ip" "test" {
+  name                = "publicIPForLB"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  allocation_method   = "Static"
+}
+
+resource "azurerm_lb" "test" {
+  name                = "loadBalancer"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+
+  frontend_ip_configuration {
+    name                 = "publicIPAddress"
+    public_ip_address_id = azurerm_public_ip.test.id
+  }
+}
+
+resource "azurerm_lb_backend_address_pool" "test" {
+  loadbalancer_id = azurerm_lb.test.id
+  name            = "BackEndAddressPool"
+}
+
+resource "azurerm_network_interface" "test" {
+  count               = 2
+  name                = "acctni${count.index}"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+
+  ip_configuration {
+    name                          = "testConfiguration"
+    subnet_id                     = azurerm_subnet.test.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_managed_disk" "test" {
+  count                = 2
+  name                 = "datadisk_existing_${count.index}"
+  location             = azurerm_resource_group.rg.location
+  resource_group_name  = azurerm_resource_group.rg.name
+  storage_account_type = "Standard_LRS"
+  create_option        = "Empty"
+  disk_size_gb         = "1023"
+}
+
+resource "azurerm_availability_set" "avset" {
+  name                         = "avset"
+  location                     = azurerm_resource_group.rg.location
+  resource_group_name          = azurerm_resource_group.rg.name
+  platform_fault_domain_count  = 2
+  platform_update_domain_count = 2
+  managed                      = true
+}
+
+resource "random_pet" "azurerm_linux_virtual_machine_name" {
+  prefix = "vm"
+}
+
+resource "azurerm_linux_virtual_machine" "test" {
+  count                 = 2
+  name                  = "${random_pet.azurerm_linux_virtual_machine_name.id}${count.index}"
+  location              = azurerm_resource_group.rg.location
+  availability_set_id   = azurerm_availability_set.avset.id
+  resource_group_name   = azurerm_resource_group.rg.name
+  network_interface_ids = [element(azurerm_network_interface.test.*.id, count.index)]
```

Comment:
Since both `count` of nic and vm are `2`, could we simplify the expression here?:

```hcl
network_interface_ids = [azurerm_network_interface.test[count.index].id]
```